bundles/core/src/main/java/com/adobe/cq/wcm/core/components/util/ComponentUtils.java

```diff
@@ -0,0 +1,148 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2020 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.util;
+
+import com.adobe.cq.wcm.core.components.internal.DataLayerConfig;
+import com.adobe.cq.wcm.core.components.internal.Utils;
+import com.adobe.cq.wcm.core.components.models.Component;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.day.cq.wcm.api.Template;
+import com.day.cq.wcm.api.components.ComponentContext;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.caconfig.ConfigurationBuilder;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Optional;
+
+import static com.adobe.cq.wcm.core.components.internal.Utils.ID_SEPARATOR;
+
+/**
+ * Utility helper functions for components.
+ */
+public final class ComponentUtils {
+
+    /**
+     * Private constructor to prevent instantiation of utility class.
+     */
+    private ComponentUtils() {
+        // NOOP
+    }
+
+    /**
+     * Check if data layer is enabled.
+     *
+     * @param resource The resource to check.
+     * @return True if data layer is enabled for this resource, false otherwise.
+     */
+    public static boolean isDataLayerEnabled(@NotNull final Resource resource) {
+        return Optional.ofNullable(resource.adaptTo(ConfigurationBuilder.class))
+            .map(builder -> builder.as(DataLayerConfig.class))
+            .map(DataLayerConfig::enabled)
+            .orElse(false);
+    }
+
+    /**
+     * Get the ID property value if set (using {@link #getPropertyId(Resource)},
+     * otherwise generate a new ID (using {@link #generateId(Resource, Page, ComponentContext)}.
+     *
+     * @param resource The resource for which to get or generate an ID.
+     * @param currentPage The current request page.
+     * @param componentContext The current component context.
+     * @return The ID property value for the specified resource, or a generated ID if not set.
+     */
+    @NotNull
+    public static String getPropertyOrGeneratedId(@NotNull final Resource resource,
```

Comment:
Does it make sense to expose the other two methods? I would rather rename this to `getId` and keep the other two methods private.

I would move `Utils#generateId(prefix, path)` here and make it public though, in case people want to reuse it.